src/Payum/Paypal/ExpressCheckout/Nvp/Action/CancelAction.php

```diff
@@ -41,9 +37,12 @@ public function execute($request)
      */
     public function supports($request)
     {
-        return
-            $request instanceof Cancel &&
-            $request->getModel() instanceof \ArrayAccess
-        ;
+        if (false == ($request instanceof Cancel && $request->getModel() instanceof \ArrayAccess)) {
+            return false;
+        }
+
+        // it must take into account only common payments, recurring payments must be cancelled by another action.
+        $model = $request->getModel();
+        return isset($model['TRANSACTIONID']);
```

Comment:
Also moving the action namespace is BC break, do you have any convention for deprecation?

I could keep the old class extending the new class and add:
```php
trigger_error('The ' . __CLASS__ . ' class is deprecated, use . NewCancelAction::class . , it will be removed in the next major version.', E_USER_DEPRECATED);
```